backend/src/v2/driver/driver.go

```diff
@@ -678,13 +678,24 @@ func resolveInputs(ctx context.Context, dag *metadata.DAG, iterationIndex *int,
 				}
 			}
 			value, hasValue := inputs.GetParameterValues()[name]
+
+			// Skip when hasValue == false, which means this parameter does not have an input value
 			if !hasValue {
-				if spec.GetDefaultValue() == nil {
-					return fmt.Errorf("input parameter %q is required", name)
-				}
-				inputs.GetParameterValues()[name] = spec.GetDefaultValue()
-				value = spec.GetDefaultValue()
+				continue
```

Comment:
I might be wrong, but wouldn't this make the two if-statements below unreachable? Should we remove the code and leave the comment there explaining the reasoning behind the decision to not replace the parameters to their default values?

Reply:
Thank you, I should have removed this.